Add API doc to logs pipelines array processor (#31339)
DataDog/documentation: content/en/logs/log_configuration/processors.md

}
```

{{% /tab %}}
{{% tab "API" %}}

> Use the [Datadog Log Pipeline API endpoint][1] with the following array processor JSON payload:

Reviewer: ```suggestion
Use the [Datadog Log Pipeline API endpoint][1] with the following array processor JSON payload:
```

```json
{
  "type": "array-processor",
  "name": "Extract Referrer URL",
  "is_enabled": true,
  "operation" : {
    "type" : "select",
    "source": "httpRequest.headers",
    "target": "referrer",
    "filter": "name:Referrer",
    "value_to_xtract": "value"
  }
}
```

| Parameter    | Type             | Required | Description                                                   |
|--------------|------------------|----------|---------------------------------------------------------------|
| `type`       | String           | Yes      | Type of the processor.                                        |
| `name`       | String           | No       | Name of the processor.                                        |

> | `is_enabled` | Boolean          | No       | If the processors is enabled or not. Default: `false`.        |

Reviewer: ```suggestion
| `is_enabled` | Boolean          | No       | Whether the processor is enabled. Default: `false`.        |
```

> | `operation.type`  | String      | Yes      | Type of array processor operation.                            |
> | `operation.source`  | String    | Yes      | Path of the array you want to select from.                    |
> | `operation.target`  | String    | Yes      | Target attribute.                                             |
> | `operation.filter`  | String    | Yes      | The first array element matching this expression is selected. |
> | `operation.value_to_extract`  | String | Yes | Attribute to read in the selected element.                  |
> | `operation.type`  | String      | Yes      | Type of array processor operation.                            |

Reviewer: ```suggestion
| `operation.type`  | String      | Yes      | Type of array processor operation.                            |
| `operation.source`  | String    | Yes      | Path of the array you want to select from.                    |
| `operation.target`  | String    | Yes      | Target attribute.                                             |
| `operation.filter`  | String    | Yes      | Expression to match an array element. The first matching element is selected. |
| `operation.value_to_extract`  | String | Yes | Attribute to read in the selected element.                  |
```

}
```
{{% /tab %}}
{{% tab "API" %}}

Use the [Datadog Log Pipeline API endpoint][1] with the following array processor JSON payload:

```json
{
  "type": "array-processor",
  "name": "Compute number of tags",
  "is_enabled": true,
  "operation" : {
    "type" : "length",
    "source": "tags",
    "target": "tagCount"
  }
}
```

| Parameter           | Type      | Required | Description                                                   |
|---------------------|-----------|----------|---------------------------------------------------------------|
| `type`              | String    | Yes      | Type of the processor.                                        |
| `name`              | String    | No       | Name of the processor.                                        |

> | `is_enabled`        | Boolean   | No       | If the processors is enabled or not. Default: `false`.        |

Reviewer: ```suggestion
| `is_enabled`        | Boolean   | No       | Whether the processor is enabled. Default: `false`.        |
```

}
```
{{% /tab %}}
{{% tab "API" %}}

Use the [Datadog Log Pipeline API endpoint][1] with the following array processor JSON payload:

```json
{
  "type": "array-processor",
  "name": "Append client IP to sourceIps",
  "is_enabled": true,
  "operation" : {
    "type" : "append",
    "source": "network.client.ip",
    "target": "sourceIps"
  }
}
```

| Parameter                    | Type       | Required | Description                                                        |
|------------------------------|------------|----------|--------------------------------------------------------------------|
| `type`                       | String     | Yes      | Type of the processor.                                             |
| `name`                       | String     | No       | Name of the processor.                                             |

> | `is_enabled`                 | Boolean    | No       | If the processors is enabled or not. Default: `false`.             |

Reviewer: ```suggestion
| `is_enabled`                 | Boolean    | No       | Whether the processor is enabled. Default: `false`.             |
```

| `operation.type`             | String     | Yes      | Type of array processor operation.                                 |
| `operation.source`           | String     | Yes      | Attribute to append.                                               |
| `operation.target`           | String     | Yes      | Array attribute to append to.                                      |

> | `operation.preserve_source`  | Boolean    | No      | Remove or preserve the remapped source element. Default: `false`.   |

Reviewer: ```suggestion
| `operation.preserve_source`  | Boolean    | No      | Whether to preserve the original source after remapping. Default: `false`.   |
```
